src/operand/reference.ts

```diff
@@ -102,6 +119,13 @@ export class Reference extends Operand {
     }
     super()
     this.key = key
+
+    const dataTypeRegex = /^.+\.\((Number|String|Date)\)$/
```

Comment:
Hopefully we can support date comparison at some point

Reply:
If it's for "at some point" maybe we shouldn't include `Date` now?

Reply:
sure, just a POC, can eventually have this implemented in this PR if agreed

Reply:
I agree, the `Date` is a bit more complex as we have to also define which format of date would be supported for this casting.

Reply:
58aaeca